app/controllers/api/service_dialogs_controller.rb

```diff
@@ -98,5 +105,11 @@ def define_service_dialog(dialog_fields, data, options = {})
     def service_dialog_ident(service_dialog)
       "Service Dialog id:#{service_dialog.id} label:'#{service_dialog.label}'"
     end
+
+    def validate_orchestration_template_dialog_create_data(data)
+      bad_attrs = []
+      ORCHESTRATION_TEMPLATE_DIALOG_REQUIRED_ATTRS.each { |attr| bad_attrs << attr if data[attr].blank? }
```

Comment:
maybe rename bad_attrs to missing_attributes ? and then do:
```ruby
missing_attributes  = ORCHESTRATION_TEMPLATE_DIALOG_REQUIRED_ATTRS - data.keys
```
what do you think ?

Reply:
and think you need to check that in data are just `ORCHESTRATION_TEMPLATE_DIALOG_REQUIRED_ATTRS ` keys, right ?

so you can do in addition 
```
invalid_attributes  =  data.keys - ORCHESTRATION_TEMPLATE_DIALOG_REQUIRED_ATTRS
raise BadRequestError if invalid_attributes.present?
```